intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/verifiers/ProductReleaseVersionVerifier.kt

```diff
@@ -0,0 +1,67 @@
+package com.jetbrains.plugin.structure.intellij.verifiers
+
+import com.jetbrains.plugin.structure.base.problems.NotNumber
+import com.jetbrains.plugin.structure.base.problems.PropertyNotSpecified
+import com.jetbrains.plugin.structure.intellij.beans.PluginBean
+import com.jetbrains.plugin.structure.intellij.problems.ReleaseVersionWrongFormat
+import com.jetbrains.plugin.structure.intellij.problems.SuspiciousReleaseVersion
+import com.jetbrains.plugin.structure.intellij.verifiers.ProductReleaseVersionVerifier.VerificationResult.Invalid
+import com.jetbrains.plugin.structure.intellij.version.ProductReleaseVersion
+
+class ProductReleaseVersionVerifier {
+  fun verify(plugin: PluginBean, descriptorPath: String, problemRegistrar: ProblemRegistrar): VerificationResult {
+    if (plugin.productDescriptor == null) return VerificationResult.NotApplicable
+
+    val releaseVersionValue = plugin.productDescriptor?.releaseVersion
+    if (releaseVersionValue.isNullOrEmpty()) {
+      return Invalid("Attribute '$RELEASE_VERSION_ATTRIBUTE_NAME' is missing").also {
+        problemRegistrar.registerProblem(PropertyNotSpecified(RELEASE_VERSION_ATTRIBUTE_NAME, descriptorPath))
+      }
+    }
+
+    return try {
+      ProductReleaseVersion.parse(releaseVersionValue).run {
+        if (isSingleDigit) {
+          Invalid("Attribute '$RELEASE_VERSION_ATTRIBUTE_NAME' must have two or more digits: '$releaseVersionValue'").also {
+            problemRegistrar.registerProblem(ReleaseVersionWrongFormat(descriptorPath, releaseVersionValue))
+          }
+        } else {
+          verifyPluginVersionAndReleaseVersionMatch(plugin, productReleaseVersion = this, descriptorPath, problemRegistrar)
+          VerificationResult.Valid(this)
+        }
+      }
+    } catch (e: NumberFormatException) {
+      Invalid("Attribute '$RELEASE_VERSION_ATTRIBUTE_NAME' is not an integer: '$releaseVersionValue'").also {
+        problemRegistrar.registerProblem(NotNumber(RELEASE_VERSION_ATTRIBUTE_NAME, descriptorPath))
+      }
+    }
+  }
+
+  private fun verifyPluginVersionAndReleaseVersionMatch(
+    plugin: PluginBean,
+    productReleaseVersion: ProductReleaseVersion,
+    descriptorPath: String,
+    problemRegistrar: ProblemRegistrar
+  ) {
+    if (plugin.pluginVersion == null) return
+
+    val commonPrefix = productReleaseVersion.major.toString().commonPrefixWith(plugin.pluginVersion)
```

Comment:
Shouldn't be `minor` component also be checked. If I'm not mistaken it allows to have releaseVersion = 20221 and version = 2022.5.2?